drivers/gpu/drm/vc4/vc4_plane.c

```diff
@@ -1943,7 +1943,7 @@ static int vc6_plane_mode_set(struct drm_plane *plane,
 
 		components_per_word = fb->format->format == DRM_FORMAT_P030 ? 24 : 32;
 		starting_offset = src_x % components_per_word;
-		fetch_count = (width + starting_offset + components_per_word - 1) /
+		fetch_count = (width + starting_offset + components_per_word) /
```

Comment:
This feels like for the common case (e.g width=1920, starting_offset=0, components_per_word=32) we will now fetch 61 "words" rather than 60. And 60 is all that is needed.

That is both increasing sdram bandwidth and the potential for reading something you shouldn't.

I think we want to make this only fetch the extra word in the cases we know it goes wrong (perhaps when `starting_offset != 0`, or even `>= components_per_word-  2`).

Note: the fetcher spec does contain details on calculating this for different formats.

Reply:
Updated to revert back to `if(!i && last_bits == 22) fetch_count++`
That still works with zooming in and out using Kodi, but avoids touching most unnecessary cases. It will need some more analysis before being able to upstream the fix.